autotest/cpp/test_ogr.cpp

```diff
@@ -2163,4 +2164,82 @@ namespace
         EXPECT_EQ(i, oFDefn.GetGeomFieldCount());
     }
 
+    // Test GDALDataset QueryLoggerFunc callback
+    TEST_F(test_ogr, GDALDatasetSetQueryLoggerFunc)
+    {
+      if( GDALGetDriverByName("GPKG") == nullptr )
+      {
+          GTEST_SKIP() << "GPKG driver missing";
+      }
+
+      GDALDatasetH ds { GDALOpenEx( "data/poly-1-feature.gpkg", GDAL_OF_VECTOR|GDAL_OF_UPDATE, nullptr, nullptr, nullptr )};
+      ASSERT_TRUE( ds );
```

Comment:
Static analyzers (Coverity Scan) will complain about memory leaks if one of the below ASSERT() triggers. So doing the following should help
```suggestion
      auto poDS = std::unique_ptr<GDALDataset>(GDALDataset::Open("data/poly-1-feature.gpkg", GDAL_OF_VECTOR|GDAL_OF_UPDATE));
      ASSERT_TRUE( poDS );
      auto hDS = GDALDataset::ToHandle(poDS);
```

Reply:
And you should actually copy the dataset to a temporary one to avoid modifying a file that is in git

Reply:
Done